### ISSUE TITLE
Analyzer hint for no return statement shouldn't apply to Null functions

### ISSUE THREAD

**Issue by leafpetersen** [dart-lang/sdk#28656]
This code

```dart
Null returnsNothing() {
}
```
Causes a hint about no return statement.  This seems unnecessary.

**Comment by lrhn** [comment]
Not to me.
A void function doesn't return anything. It can omit the return or use a return statement with no expression.
A non-void function returns something. It should use a return with an expression.

Trying to treat `return;` (or nothing) the same as `return null;` removes this distinction between a procedure/void function and an actual non-void function. Then you can argue that you should be able to use `return;` in all functions, as a shorthand for `return null;`, and I don't want that.

Basically I want `return;` and exit-with-no-return to act like an implicit return of type `void`, not of type `Null`.

**Comment by srawlins** [comment]
I agree, `void` has since become a very common/standard return type. I vote Analyzer should continue marking this as `missing_return`.

**Comment by srawlins** [comment]
We've added new diagnostics (Hint) for a missing return on a nullable return type as well.

**Comment by srawlins** [comment]
We've done a lot of work in this area for null safety, so I'll close this out.
